main.go

```diff
@@ -218,11 +244,27 @@ func main() {
 		os.Exit(1)
 	}
 
+	multiclusterengineList := &backplanev1.MultiClusterEngineList{}
+	err = uncachedClient.List(context.TODO(), multiclusterengineList)
+	if err != nil {
+		setupLog.Error(err, "Could not set List multicluster engines")
+		os.Exit(1)
+	}
+
+	if len(multiclusterengineList.Items) == 0 {
+		err = upgradeableCondition.Set(ctx, metav1.ConditionTrue, utils.UpgradeableAllowReason, utils.UpgradeableAllowMessage)
```

Comment:
I think it would be better to identify what the condition should be, and then set it only once in main. Currently if there are no deployed MCE this will set the operatorcondition to Upgradeable->False and then immediately after set it to True.

Reply:
@JakobGray but isn't that the discussion we were having this morning? The condition should be set to false until the readiness probe is set, before defaulting to true. Otherwise, we can just default it to true until the reconciler runs to change it

Reply:
If the operatorCondition is considered upgradable because no MCE's exist, then setting it as non-upgradable before changing it to upgradable doesn't change things. Everything in main happens before the readiness probe is active. This code will still work and since the CRDs are processed in between the two calls it's unlikely for a conflict to occur and crash the pod so I'll mark this as resolved.